src/components/Modal/Modal.stories.tsx

```diff
@@ -0,0 +1,218 @@
+            control: { type: "select" },
+        },
+        title: {
+            table: {
+                category: "Header",
+            },
+            name: "Title",
+            defaultValue: "Awesome modal",
+            control: { type: "text" },
+        },
+        leadText: {
+            table: {
+                category: "Header",
+            },
+            name: "Lead Text",
+            defaultValue: "The Home for Brand Management",
+            control: { type: "text" },
+        },
+        decorator: {
+            table: {
+                category: "Header",
+            },
+            name: "Decorator",
+            options: ["None", "Icons", "Academy", "Audio"],
+            mapping: {
+                None: null,
+                Icons: <IconIcons key="iconIcons" />,
+                Academy: <IconAcademy key="iconAcademy" />,
+                Audio: <IconAudio key="iconAudio" />,
+            },
+            defaultValue: "Icons",
+            control: { type: "select" },
+        },
+        variant: {
+            table: {
+                category: "Header",
+            },
+            name: "Variant",
+            options: ["None", ...Object.values(ModalHeaderVariant)],
+            mapping: { None: null, ...Object.values(ModalHeaderVariant) },
+            defaultValue: ModalHeaderVariant.Default,
+            control: { type: "select" },
+        },
+    },
+    decorators: [
+        (Story) => (
+            <OverlayProvider>
+                <OverlayContainer>
+                    <Story />
+                </OverlayContainer>
+            </OverlayProvider>
+        ),
+    ],
+} as Meta;
+
+export const Default: Story<ModalProps & ModalVisualProps & ModalHeaderProps> = (args) => {
+    const [input, setInput] = useState("");
+
+    const state = useOverlayTriggerState({});
+
+    return (
+        <>
+            <Button onClick={() => state.open()}>Open Modal</Button>
+            {state.isOpen && (
+                <Modal
+                    visual={{
+                        pattern: args.pattern,
+                        foregroundColor: args.foregroundColor,
+                    }}
+                    header={
+                        <ModalHeader
+                            title={args.title}
+                            leadText={args.leadText}
+                            decorator={args.decorator}
+                            variant={args.variant}
+                        />
+                    }
+                    footer={
+                        <ModalFooter
+                            buttons={[
+                                { label: "Okay", action: () => action("click") },
+                                {
+                                    label: "Cancel",
+                                    action: () => state.close(),
+                                },
+                            ]}
+                        />
+                    }
```

Comment:
@SamuelAlev do you think for the header, footer and body we should just pass props instead of full components and automatically wrap in the ModalHeader, ModalFooter and ModalBody components respectively?